### DISCUSSION TITLE
Attaching files in an e-mail?

### DISCUSSION THREAD

**Discussion by wjcagle** [pimalaya/himalaya#211] [General]
I'm still new to HImalay, and so far I've been unable to find this. I see you can download attachments, but is there a way to send an attachment with Himalaya when writing an e-mail?

**Comment by soywod** [comment]
Yes there is, but I'm not sure about the stability of this feature. It should be better managed in the `v0.5.0`. For now, attachments can be attached with the `-a|--attachment` param with the `write` command. Sth like `himalaya write --attachment /abs/path/to/file`.

**Comment by gl-yziquel** [comment]
Seems like this option is not available in my 0.8.4-beta version of himalaya.

**Reply by soywod** [reply]
If I don't mistake, the `v0.8.4` already contains the MML language, which is a language that allows you to write MIME Messages in a simpler way. Have a look at <https://git.sr.ht/~soywod/pimalaya/tree/master/item/mml/README.md>.

**Reply by soywod** [reply]
To summarize, there is no more CLI option, instead you can add attachments directly from your email message using the MML language.

**Reply by gl-yziquel** [reply]
In the documentation you gave, you see things like the following:

```console
<#part disposition=attachment filename=/path/to/attachment.ext><#/part>
```

This seems to be misleading as what works for me on 0.84 is:

```console
<#part disposition=attachment filename=/path/to/attachment.ext></#part>
```

Notice the position of the '/'.

It would seem to me wise to correct the docs.

**Reply by gl-yziquel** [reply]
Well, well, well... Just had a look at mml in https://www.gnu.org/software/emacs/manual/html_node/emacs-mime/Simple-MML-Example.html

The <#...><#/...> is the correct format for mml. Not <#...></#...>.

So the documentation is correct, but the code seems to be wrong.

**Reply by soywod** [reply]
Really strange: the CLI `v0.8.4` used [`email-lib@=0.14.0`](https://github.com/soywod/himalaya/blob/fff82498baf9ad8168e86728dd7e546306fd00f1/Cargo.toml#L48), which has been released from the commit [2c13a](https://git.sr.ht/~soywod/pimalaya/commit/2c13a7bdddeb9f266ff8bea080465036d5b8d9e3), and the end tag seems correct [in the code](https://git.sr.ht/~soywod/pimalaya/tree/2c13a7bdddeb9f266ff8bea080465036d5b8d9e3/item/email-tpl/src/mml/parsers/parts.rs#L10). I invite you to use the `v0.9.0` or to wait for the `v1` (in both cases you will have breaking changes).

**Reply by gl-yziquel** [reply]
The v0.9.0 version unfortunately does not spawn up my editor with `himalaya write`. I haven't investigated why yet.

**Reply by soywod** [reply]
This issue has been identified [here](https://todo.sr.ht/~soywod/pimalaya/150) and I have a fix that will be released with the v1. The workaround is to use a GUI editor (for example, if you use vim, then use gvim instead). Sorry for all those bugs, I really hope for the v1 to start consolidating the code base!

**Reply by gl-yziquel** [reply]
A GUI editor ? As Elie Semoun said in Kaamelott... [Hérétique, au bûcher !!](https://www.youtube.com/watch?v=iwxswypnRec)

(I can live with bugs...)

**Reply by soywod** [reply]
Emacs au bûcher also then? :smile:

**Reply by gl-yziquel** [reply]
I use spacemacs in a terminal over ssh without X. A black terminal with green characters is all one needs in life.
